### ISSUE TITLE
Gradients get NaN when using fp16 in Sparse RCNN

### ISSUE THREAD

**Issue by Mingfeng-Wang** [open-mmlab/mmdetection#6738]
I noticed other models` config with fp16 is offered except Sparse RCNN. Do you find this problem as well?
It works normally when I test Faster RCNN with fp16.

My setting is:
Single GPU 3090.
loss_scale = 512.
1x_r50_fpn_100pro

**Comment by RangiLyu** [comment]
Sorry, Sparse RCNN does not support fp16 yet.

**Comment by Mingfeng-Wang** [comment]
@RangiLyu 

Would mind giving some details about why Sparse RCNN does not support fp16 yet? Human resources or working on it or something?

I am considering working on it probably.

Thanks

**Comment by ZwwWayne** [comment]
Do you change the learning rate accordingly?

**Comment by Mingfeng-Wang** [comment]
> Do you change the learning rate accordingly?
> 
> You might need to divide the LR by 8 if you only use 1 GPU.

The batch size is all 16. I don't quite understand why I need to change learning rate. Could you explain for me.

Thanks a lot.

**Comment by RangiLyu** [comment]
Hi, we refactored the fp16 training in mmdet v3.0. https://github.com/open-mmlab/mmdetection/tree/3.x

You can use the following command to train sparse rcnn with fp16:
```
python tools/train.py configs/sparse_rcnn/sparse-rcnn_r50_fpn_1x_coco.py --amp
```

Welcome to try it and feel free to open an issue if you met some problem with the 3.x version.

**Comment by zylo117** [comment]
amp fp16 may cause overflow in some avg pooling during summing up. mmdetection clearly didn't consider this so sometimes nan happens in amp mode.
